pkg/cmd/issue/comment/comment.go

```diff
@@ -0,0 +1,215 @@
+package comment
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/issue/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/cli/cli/pkg/surveyext"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type CommentOptions struct {
+	HttpClient func() (*http.Client, error)
+	Config     func() (config.Config, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Edit       func(string) (string, error)
+
+	SelectorArg string
+	Interactive bool
+	InputType   int
+	Body        string
+}
+
+const (
+	inline = iota
+	editor
+	web
+)
+
+func NewCmdComment(f *cmdutil.Factory, runF func(*CommentOptions) error) *cobra.Command {
+	opts := &CommentOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Config:     f.Config,
+		Edit: func(editorCommand string) (string, error) {
+			return surveyext.Edit(editorCommand, "*.md", "", f.IOStreams.In, f.IOStreams.Out, f.IOStreams.ErrOut, nil)
+		},
+	}
+
+	var webMode bool
+	var editorMode bool
+
+	cmd := &cobra.Command{
+		Use:   "comment {<number> | <url>}",
+		Short: "Create a new issue comment",
+		Example: heredoc.Doc(`
+			$ gh issue comment 22 --body "I found a bug. Nothing works"
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+			opts.SelectorArg = args[0]
+			opts.Interactive = !(opts.Body != "" || webMode || editorMode)
+
+			if !opts.IO.CanPrompt() && opts.Interactive {
+				return &cmdutil.FlagError{Err: errors.New("--body, --editor, or --web required when not running interactively")}
+			}
+
+			if !opts.Interactive {
+				if opts.Body != "" && !webMode && !editorMode {
+					opts.InputType = inline
+				} else if opts.Body == "" && webMode && !editorMode {
+					opts.InputType = web
+				} else if opts.Body == "" && !webMode && editorMode {
+					opts.InputType = editor
```

Comment:
Maintaining all permutations of these conditions feels like a code smell to me. Check https://github.com/cli/cli/blob/f4152454f2e2038e87b2516bcd419660945c501f/pkg/cmd/pr/merge/merge.go#L71-L88 to see a cleaner ad-hoc pattern of enforcing mutually-exclusive flags.